### PR TITLE
feat: using 'init.defaultBranch' when initializing git repo

### PR DESCRIPTION
**Pull request by Mvmo**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog
Introduced new function `getDefaultBranch()` in `./cli/src/helpers/git.ts`, which either returns the default branch name configured on the users computer or `main`.
Said function is called everytime the user decides to automatically initialize a git repository to set the default branch

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

File: .changeset/orange-books-fix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"create-t3-app": patch
+---
+
+Fixes fallback branch if init.defaultBranch is not set
```
Comment (c-ehrlich) [inline .changeset/orange-books-fix.md:5]: There should only be one changeset for a PR

**Review by c-ehrlich — COMMENTED** [review]
Thanks for the PR.

How common is it for people to:
1. have `init.defaultBranch` configured?
2. have it configured to anything other than `main`?

I'm leaning towards merging this either way because it's not a lot of logic, but want to hear opinions from other maintainers first.

File: .changeset/orange-books-fix.md (hunk 1 above)
Comment (Mvmo) [reply .changeset/orange-books-fix.md]: I've removed the changelog commit

**Comment by t3dotgg** [comment]
Good change imo

**Review by c-ehrlich — APPROVED** [review]
👍

**Pull request merged**